resources/assets/js/quick-links.js

```diff
@@ -20,3 +21,18 @@ function initQuickLinksForm() {
         });
     });
 }
+
+function showFilterField() {
+    selector = $('#entity-selector');
+    if (selector.val() != '') {
```

Comment:
Missing a test to make sure the field exists before testing it's value, to avoid having scripts running when not needed, slowing down the browser.